include/mbedtls/pk.h

```diff
@@ -361,24 +361,28 @@ int mbedtls_pk_setup(mbedtls_pk_context *ctx, const mbedtls_pk_info_t *info);
 /**
  * \brief           Initialize a PK context to wrap a PSA key.
  *
- * \note            This function replaces mbedtls_pk_setup() for contexts
- *                  that wrap a (possibly opaque) PSA key instead of
- *                  storing and manipulating the key material directly.
+ *                  This function helps creating a PK context which wraps a
+ *                  PSA key. The PSA wrapped key must:
+ *                  * remain valid as long as the wrapping PK context is in use,
+ *                    that is at least between the point this function is
+ *                    called and the point mbedtls_pk_free() is called on this
+ *                    context;
+ *                  * be a key pair;
+ *                  * be an EC or RSA type (DH is not suported in PK module).
+ *
+ *                  Under the hood PSA functions are used to perform the required
+ *                  operations and, based on the key type, used algorithms will be:
+ *                  * EC:
+ *                      * verify: #PSA_ALG_ECDSA_ANY;
+ *                      * sign: try both deterministic and non-deterministic ECDSA.
+ *                  * RSA:
+ *                      * sign: #PSA_ALG_RSA_PKCS1V15_SIGN();
+ *                      * decrypt: #PSA_ALG_RSA_PKCS1V15_CRYPT.
```

Comment:
This looks like a bug similar to https://github.com/Mbed-TLS/mbedtls/issues/8824, though here the wrong algorithm will lead to a runtime error. We should probably extract the algorithm from the policy like `mbedtls_pk_copy_from_psa`.

Reply:
Right, this is what I had in mind too, but as I mentioned in the [comment above](https://github.com/Mbed-TLS/mbedtls/pull/8937#issuecomment-2003450603) doesn't this affect library's code and therefore impact with the code freeze?

Reply:
Code freeze is this afternoon, this pull request isn't going to be reviewed in time anyway. This is a bug we'll fix in 3.6.1.

Reply:
I think we can just document it as a current limitation. Also, we should distinguish by function:
- `pk_sign()` will always try `PKCS1V15_SIGN`, and fail if that's not allowed by the key policy;
- `pk_sign_ext()` will try the algorithm corresponding to its `pk_type` argument and fail if that's not allowed by the key policy.

So, people who want to do PSS signatures with Opaque RSA keys currently need to use `sign_ext()`.

In the future, we can expand `pk_sign()` to Do What I Mean - but that would be an improvement. IMO the current situation is more a limitation than a bug. In particular, I don't see it as urgent to change before code freeze.